ethcore/src/engines/authority_round.rs

```diff
@@ -209,7 +207,9 @@ impl Engine for AuthorityRound {
 	}
 
 	fn populate_from_parent(&self, header: &mut Header, parent: &Header, gas_floor_target: U256, _gas_ceil_target: U256) {
-		header.set_difficulty(parent.difficulty().clone());
+		// Chain scoring: total weight is sqrt(U256::max_value())*height - step
+		let new_difficulty = U256::from(U128::max_value()) + header_step(parent).expect("Header has been verified; qed").into() - self.step.load(AtomicOrdering::SeqCst).into();
```

Comment:
But no need to worry, releasing billion blocks every second it will last us until the death of the galaxy many times anyway.